### PR TITLE
CYSPA-25: Remove Abroad Exclusion changes.

### PR DESCRIPTION
**Pull request by rambabu-posa**

### CODE REVIEW

File: test/uk/gov/hmrc/statepension/services/StatePensionServiceSpec.scala [hunk 1]
```diff
@@ -1186,36 +1191,40 @@ class StatePensionServiceSpec extends StatePensionUnitSpec with OneAppPerSuite w
...
 
-      lazy val exclusionF: Future[StatePensionExclusion] = service.getStatement(generateNino()).left.get
+      lazy val statePensionF: Future[StatePension] = service.getStatement(generateNino()).right.get
 
-      "return abroad" in {
-        whenReady(exclusionF) { exclusion =>
-          exclusion.exclusionReasons shouldBe List(Exclusion.Abroad)
+      "NOT return abroad" in {
+        whenReady(statePensionF) { statePension =>
+          statePension should not be (Nil)
```
Comment (callumcodes) [inline test/uk/gov/hmrc/statepension/services/StatePensionServiceSpec.scala]: I'm not sure if this is testing anything

File: public/api/conf/1.0/schemas/state-pension.json [hunk 2]
```diff
@@ -230,6 +230,11 @@
       "type": "boolean",
       "description": "Whether the customer has a \u2018reduced rate election\u2019 in force at the beginning of the tax year 35 years before the tax year of State Pension age",
       "example": "true"
+    },
+    "abroadAutoCredits": {
+      "type": "boolean",
+      "description": "Whether the customer is male, reached State Pension age before 6 October 2018 and has an overseas address",
```
Comment (callumcodes) [inline public/api/conf/1.0/schemas/state-pension.json]: This does not explain the context of the of the flag and what it means for the consumer.

File: test/uk/gov/hmrc/statepension/services/StatePensionServiceSpec.scala (hunk 1 above)
Comment (rambabu-posa) [reply test/uk/gov/hmrc/statepension/services/StatePensionServiceSpec.scala]: Do you want me test a SP object something like
statePension should be 
(StatePension(Matchers.any(),Matchers.any(),Matchers.any(),Matchers.any(),Matchers.any(),Matchers.any(),Matchers.any(),Matchers.any(),Matchers.any(),Matchers.any()))

or Check values??

File: test/uk/gov/hmrc/statepension/services/ExclusionServiceSpec.scala [hunk 3]
```diff
@@ -106,7 +106,7 @@ class ExclusionServiceSpec extends StatePensionUnitSpec {
         "the user is abroad" when {
```
[the remark this answers is not in the document]
Comment (rambabu-posa) [reply test/uk/gov/hmrc/statepension/services/ExclusionServiceSpec.scala]: Is it good to remove whole Abroad section from this service?? Will also remove associated variables like isAbroad, sex etc.
"the overseas male auto credits (abroad) exclusion criteria is met" when {
    ......
}

File: test/uk/gov/hmrc/statepension/domain/nps/StatePensionSpec.scala [hunk 4]
```diff
@@ -138,6 +140,16 @@ class StatePensionSpec extends StatePensionUnitSpec {
     }
   }
 
+  "Abroad customer" should {
```
Comment (callumcodes) [inline test/uk/gov/hmrc/statepension/domain/nps/StatePensionSpec.scala]: Can you use the term abroad/overseas auto credits or something, it's very confusing as there is logic in the service for just abroad people, which has nothing to do with the exclusion.

File: test/uk/gov/hmrc/statepension/services/ExclusionServiceSpec.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (callumcodes) [reply test/uk/gov/hmrc/statepension/services/ExclusionServiceSpec.scala]: Remove this block

[the remark this answers is not in the document]
Comment (callumcodes) [reply test/uk/gov/hmrc/statepension/services/ExclusionServiceSpec.scala]: Delete all the unecessary code and variables yes, that's refactoring.

File: test/uk/gov/hmrc/statepension/services/StatePensionServiceSpec.scala (hunk 1 above)
Comment (callumcodes) [reply test/uk/gov/hmrc/statepension/services/StatePensionServiceSpec.scala]: Check the values

File: public/api/conf/1.0/schemas/state-pension.json (hunk 2 above)
Comment (callumcodes) [reply public/api/conf/1.0/schemas/state-pension.json]: This should explain to the API consumer what to do (exclude them) and why they have to do so.

**Pull request merged**